### PR TITLE
[PEA] mark monitor inputs as escaped

### PR DESCRIPTION
**Pull request by caojoshua**
This passes `compiler/c2/TestStressRecompilation.java`.

This should help us avoid issues relating to monitors. We can revisit PEA support for monitors in the future.

### CODE REVIEW

**Comment by caojoshua** [comment]
6 failures in jtreg hotspot tier1. Only Gtest and chlsdb test.

```
gtest/NMTGtests.java#nmt-detail
gtest/NMTGtests.java#nmt-off
gtest/NMTGtests.java#nmt-summary
serviceability/sa/ClhsdbCDSCore.java
serviceability/sa/ClhsdbCDSJstackPrintAll.java
serviceability/sa/ClhsdbPrintAll.java
```

File: src/hotspot/share/opto/locknode.cpp [hunk 1]
```diff
@@ -189,6 +189,17 @@ void Parse::do_monitor_enter() {
   // the monitor object is not part of debug info expression stack
   pop();
 
+  if (DoPartialEscapeAnalysis) {
+    // TODO: PEA can support monitors
+    PartialEscapeAnalysis* pea = PEA();
+    PEAState& state = jvms()->alloc_state();
+    ObjID id;
+
+    if ((id = pea->is_alias(obj)) && state.get_object_state(id)->is_virtual()) {
```
Comment (navyxliu) [inline src/hotspot/share/opto/locknode.cpp]: we can't check is_virtual() like this.  I encountered a bug before. 
now **aliases** is a global data structure, or it's flow-insensitive. In contrast, Allocation state is still flow-sensitive. It's possible that 'obj' is an alias but the object doesn't exist in current basic block. 

you can check is_virtual() using this expression:

```java
state.as_virtual(pea, obj) != nullptr
```

File: src/hotspot/share/opto/locknode.cpp [hunk 2]
```diff
@@ -189,6 +189,17 @@ void Parse::do_monitor_enter() {
   // the monitor object is not part of debug info expression stack
   pop();
 
+  if (DoPartialEscapeAnalysis) {
+    // TODO: PEA can support monitors
+    PartialEscapeAnalysis* pea = PEA();
+    PEAState& state = jvms()->alloc_state();
+    ObjID id;
```
Comment (navyxliu) [inline src/hotspot/share/opto/locknode.cpp]: id is not useful anymore.

**Review by navyxliu — APPROVED** [review]

File: src/hotspot/share/opto/locknode.cpp (hunk 2 above)
Comment (navyxliu) [reply src/hotspot/share/opto/locknode.cpp]: this doesn't require another review. you can merge it after update.

**Pull request merged**